mtags/src/main/scala-2/scala/meta/internal/pc/SemanticdbTextDocumentProvider.scala

```diff
@@ -48,7 +49,9 @@ class SemanticdbTextDocumentProvider(
     } else {
       None
     }
-    val document = unit.toTextDocument(explicitDialect)
+    // we recalculate md5, since there seems to be issue with newlines sometimes
```

Comment:
What problems were caused by this newlines problem?

Reply:
Different md5 results produced by default when no newline was at the end. Not sure exactly why.

Reply:
It makes sense that inserting a character changes the document hash, but I still don't see when the problem starts. When is this newline inserted? `unit.toTextDocument` has `md5 = unit.source.toMD5`, so it indicates that file content is changed after compilation unit is created. Is this vscode problematic behavior?

Reply:
> When is this newline inserted?

I don't know :D That is the reason, and I am not even sure if it's a newline. Just that on the same text we were calculating two different md5 results.

Reply:
Do you have steps for reproduction? It bothers me a lot :c

Reply:
When I create worksheet with last statement without a newline this started popping up. It's not strickly related tot his PR so I can revert the change

Reply:
It's okay, there is no need for reverting. I just want to investigate it ;)